File: Calculator/Java/Cli/src/test/java/com/portfolio/calculator/CalculatorTest.java
package com.portfolio.calculator;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertThrows;

class CalculatorTest {

    @Test
    void addReturnsCorrectSum() {
        assertEquals(5, Calculator.calculate(2, 3, "add"));
        assertEquals(0, Calculator.calculate(-1, 1, "add"));
        assertEquals(0, Calculator.calculate(0, 0, "add"));
    }

    @Test
    void subtractReturnsCorrectDifference() {
        assertEquals(2, Calculator.calculate(5, 3, "subtract"));
        assertEquals(-5, Calculator.calculate(0, 5, "subtract"));
        assertEquals(0, Calculator.calculate(-1, -1, "subtract"));
    }

    @Test
    void multiplyReturnsCorrectProduct() {
        assertEquals(6, Calculator.calculate(2, 3, "multiply"));
        assertEquals(0, Calculator.calculate(0, 5, "multiply"));
        assertEquals(-6, Calculator.calculate(-2, 3, "multiply"));
    }

    @Test
    void divideReturnsCorrectQuotient() {
        assertEquals(2, Calculator.calculate(6, 3, "divide"));
        assertEquals(2.5, Calculator.calculate(5, 2, "divide"));
        assertEquals(0, Calculator.calculate(0, 5, "divide"));
    }

    @Test
    void divideByZeroThrows() {
        IllegalArgumentException ex = assertThrows(IllegalArgumentException.class, () -> Calculator.calculate(5, 0, "divide"));
        assertEquals("Cannot divide by zero", ex.getMessage());
    }

    @Test
    void invalidOperatorThrows() {
        IllegalArgumentException ex = assertThrows(IllegalArgumentException.class, () -> Calculator.calculate(2, 3, "power"));
        assertEquals("Invalid operator", ex.getMessage());
    }
}

File: Calculator/Elixir/Web/Phoenix/src/test/calculator_test.exs
defmodule CalculatorTest do
  use ExUnit.Case, async: true

  test "add" do
    assert Calculator.calculate(2, 3, "add") == {:ok, 5}
  end

  test "subtract" do
    assert Calculator.calculate(10, 4, "subtract") == {:ok, 6}
  end

  test "multiply" do
    assert Calculator.calculate(6, 7, "multiply") == {:ok, 42}
  end

  test "divide" do
    assert Calculator.calculate(10, 4, "divide") == {:ok, 2.5}
  end

  test "division by zero returns error" do
    assert Calculator.calculate(10, 0, "divide") == {:error, "Cannot divide by zero"}
  end

  test "invalid operator returns error" do
    assert Calculator.calculate(10, 4, "modulo") == {:error, "Invalid operator"}
  end
end
